src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -1550,6 +1552,47 @@ export function convertPerTexelComponentToResultFormat(
   return out;
 }
 
+function swizzleComponentToTexelComponent(
+  src: PerTexelComponent<number>,
+  component: GPUComponentSwizzle
+): number {
+  switch (component) {
+    case '0':
+      return 0;
+    case '1':
+      return 1;
+    case 'r':
+      return src.R!;
+    case 'g':
+      return src.G!;
+    case 'b':
+      return src.B!;
+    case 'a':
+      return src.A!;
+  }
+}
+
+export function swizzleTexel(
+  src: PerTexelComponent<number>,
+  swizzle: GPUTextureComponentSwizzle | undefined
+): PerTexelComponent<number> {
+  swizzle = swizzle ?? 'rgba';
+  return {
+    R: swizzle[0]
```

Comment:
old code but: How can `swizzle[0]` be falsy? If the string is too short? Maybe this condition is obsolete after the API redesign?